projects/shared-modules/hms-video/components/hms-video/conference/conference.component.ts

```diff
@@ -174,6 +175,8 @@ export class ConferenceComponent implements OnInit, OnDestroy {
       hmsActions.setVideoSettings({ deviceId: this.selectedVideoDeviceId });
       hmsActions.setLocalAudioEnabled(this.isAudioEnabled);
       hmsActions.setLocalVideoEnabled(this.isVideoEnabled);
+
+      this.subscribeToMediaDevices();
```

Comment:
When this is called, then the next time it will go into else. What if the user is joining the stage for the 2nd time, wouldn't we be pushing the same combineLatest subscription again into the list of subscriptions. 
As in this.localMediaService.audio, video, etc will get resubscribed in the conference component.

Reply:
Yes we would be pushing it every time, but the problem is that the subscription actually fires before the role change happens

Reply:
One way to not resubscribe is maintaining a variable to check if we are already subscribed